File: src/SampleCRM.Web/Services/TaxTypesService.cs
using OpenRiaServices.DomainServices.Hosting;
using OpenRiaServices.DomainServices.Server;
using SampleCRM.Web.Attributes;
using SampleCRM.Web.Models;
using System.Data.Entity.Migrations;
using System.Linq;

namespace SampleCRM.Web
{
    [EnableClientAccess]
    public class TaxTypesService : SampleCRMService
    {
        [Query]
        public IQueryable<TaxType> GetTaxTypes() => _context.TaxTypes.OrderBy(t => t.Name);

        public TaxType GetTaxTypeById(long taxTypeId) => 
            GetTaxTypes().SingleOrDefault(x => x.TaxTypeID == taxTypeId);

        [Delete]
        [RestrictAccessReadonlyMode]
        public void DeleteTaxTypes(TaxType taxType)
        {
            _context.TaxTypes.Remove(taxType);
        }

        [Insert]
        [RestrictAccessReadonlyMode]
        public void InsertTaxTypes(TaxType taxType)
        {
            _context.TaxTypes.AddOrUpdate(taxType);
        }

        [Update]
        [RestrictAccessReadonlyMode]
        public void UpdateTaxTypes(TaxType taxType)
        {
            _context.TaxTypes.AddOrUpdate(taxType);
        }
    }
}

File: src/SampleCRM.Web/Services/TaxTypesService.vb
Imports OpenRiaServices.DomainServices.Hosting
Imports OpenRiaServices.DomainServices.Server
Imports SampleCRM.Web.Attributes
Imports System.Data.Entity.Migrations
Imports System.Linq

Namespace SampleCRM.Web
    <EnableClientAccess>
    Public Class TaxTypesService
        Inherits SampleCRMService

        <Query>
        Public Function GetTaxTypes() As IQueryable(Of TaxTypes)
            Return _context.TaxTypes
        End Function

        Public Function GetTaxTypeById(ByVal taxTypeId As Long) As TaxTypes
            Return _context.TaxTypes.SingleOrDefault(Function(x) x.TaxTypeID = taxTypeId)
        End Function

        <Delete>
        <RestrictAccessDeveloperMode>
        Public Sub DeleteTaxTypes(ByVal taxType As TaxTypes)
            _context.TaxTypes.Remove(taxType)
        End Sub

        <Insert>
        <RestrictAccessDeveloperMode>
        Public Sub InsertTaxTypes(ByVal taxType As TaxTypes)
            _context.TaxTypes.AddOrUpdate(taxType)
        End Sub

        <Update>
        <RestrictAccessDeveloperMode>
        Public Sub UpdateTaxTypes(ByVal taxType As TaxTypes)
            _context.TaxTypes.AddOrUpdate(taxType)
        End Sub
    End Class
End Namespace
